### PR TITLE
Add Match Creation and Dissolve Mutations

### PR DESCRIPTION
**Pull request by Jonasdoubleyou**
https://github.com/corona-school/project-user/issues/346

### CODE REVIEW

File: graphql/pupil/mutations.ts [hunk 1]
```diff
@@ -58,4 +60,25 @@ export class MutatePupilResolver {
...
+
+    @Mutation(returns => Boolean)
+    @Authorized(Role.ADMIN)
+    async pupilCreateMatchRequest(@Arg("pupilId") pupilId: number, @Ctx() context: GraphQLContext): Promise<boolean> {
+        const pupil = await getPupil(pupilId);
+        const isAdmin = context.user.roles.includes(Role.ADMIN);
+
+        await createMatchRequest(pupil, isAdmin);
+
+        return true;
```
Comment (lsc0498) [inline graphql/pupil/mutations.ts:72]: What happens if `createMatchRequest` doesn't go through?

**Review by lsc0498 — APPROVED** [review]
Looks pretty good! 👍 Sorry for the delay by the way🙈

File: graphql/pupil/mutations.ts (hunk 1 above)
Comment (Jonasdoubleyou) [reply graphql/pupil/mutations.ts:72]: Then it throws an error back to the caller (e.g. the Retool) which can show an error message to the user. Actually I wanted to return nothing here (a "void method"), but unfortunately a Void type is [not supported](https://stackoverflow.com/questions/44737043/is-it-possible-to-not-return-any-data-when-using-a-graphql-mutation). Thus I chose to return `true` which is somewhat similar. I'll add that to the documentation somewhen 😅

Comment (lsc0498) [reply graphql/pupil/mutations.ts:72]: Alright, it's fine. I just found the `return true;` a little suspicious 😅

Comment (Jonasdoubleyou) [reply graphql/pupil/mutations.ts:72]: Yeah, would also prefer if TypeGraphQL would support it ...

**Pull request merged**